lms/djangoapps/discussion/static/discussion/js/spec/views/discussion_search_view_spec.js

```diff
@@ -0,0 +1,36 @@
+define([
+    'jquery',
+    'edx-ui-toolkit/js/utils/constants',
+    'discussion/js/views/discussion_search_view'
+],
+    function($, constants, DiscussionSearchView) {
+        'use strict';
+
+        describe('DiscussionSearchView', function() {
+            var view;
+            beforeEach(function() {
+                setFixtures('<div class="search-container"></div>');
+                view = new DiscussionSearchView({
+                    el: $('.search-container'),
+                    threadListView: {
+                        performSearch: jasmine.createSpy()
+                    }
+                }).render();
+            });
+
+            describe('Search events', function() {
+                it('perform search when enter pressed inside search textfield', function() {
+                    view.$el.find('input.search').trigger($.Event('keydown', {
+                        which: constants.keyCodes.enter
+                    }));
+                    expect(view.threadListView.performSearch).toHaveBeenCalled();
+                });
+
+                it('perform search when search icon is clicked', function() {
+                    view.$el.find('button.search').click();
+                    expect(view.threadListView.performSearch).toHaveBeenCalled();
+                });
+            });
```

Comment:
It would be good to add some tests around the clearing of the search box, as well as what happens when the user is in a search and then clicks to "All Topics".